src/parquet/column/reader.cc

```diff
@@ -33,7 +33,10 @@ ColumnReader::ColumnReader(const ColumnDescriptor* descr,
   : descr_(descr),
     pager_(std::move(pager)),
     num_buffered_values_(0),
-    num_decoded_values_(0) {}
+    num_decoded_values_(0) {
+      repetition_level_decoder_.Init(descr->max_repetition_level());
+      definition_level_decoder_.Init(descr->max_definition_level());
```

Comment:
I prefer to have clean semantics for the decoder without bothering about how the upper layers call `Reset` and  `Decode` for various cases. The Reset should not be bothered about creating an instance. It should simply re-initialize the state of the decoder.  Similarly the `Decode` should return values only if there are values to decode. Else it will simply return zero elements.